internal/resultstore/resultstore.go

```diff
@@ -233,17 +224,56 @@ func (rs *ResultStore) SaveWithFilename(ctx context.Context, p Pkg, filename str
 	}
 
 	return nil
+
 }
 
-// Save marshals the given analysis results to JSON and writes them to
-// the bucket with a default filename (key). If p is non-nil and has a
-// version specified, the default filename is <version>.json.
+// DefaultFilename returns the basename (i.e. without directory-like prefixes) of the default filename (key)
+// used to store results. If p is non-nil and has a version specified, the default filename is <version>.json.
 // Otherwise, it is "results.json".
-func (rs *ResultStore) Save(ctx context.Context, p Pkg, analysis interface{}) error {
-	filename := "results.json"
+func DefaultFilename(p Pkg) string {
 	if p != nil && p.Version() != "" {
-		filename = p.Version() + ".json"
+		return p.Version() + ".json"
+	}
+	return "results.json"
+}
+
+// SaveAnalysis wraps the analysis object with the DynamicAnalysisRecord struct and saves it to the bucket
+// using saveWithFilename. If filename is empty, a default filename (chosen using DefaultFilename) is used.
+func (rs *ResultStore) SaveAnalysis(ctx context.Context, p Pkg, analysis any, filename string) error {
```

Comment:
Is this worth calling `SaveDynamicAnalysis`?

Reply:
I had this before but then I realised that file writes and execution log also use it... but I guess it's still dynamic analysis so happy to rename